src/core/maps/grid_map.h

```diff
@@ -9,19 +9,47 @@
 #include "../geometry_utils.h"
 
 class GridMap {
-public:
+private: //flag constants
+  const int RESIZE_VERT    = 0;
+  const int RESIZE_HORZ    = 1;
+  const int RESIZE_DIM_BIT = 1;
+
+  const int RESIZE_FWD     = 0;
+  const int RESIZE_BWD     = 1;
+  const int RESIZE_DIR_BIT = 0;
+#define RESIZE_DIR_BUILD(DIM,DIM_BIT,DIR,DIR_BIT) \
+  ((DIM) << (DIM_BIT)) | ((DIR) << (DIR_BIT))
+  const int RESIZE_UP    = RESIZE_DIR_BUILD(RESIZE_VERT,RESIZE_DIM_BIT,
```

Comment:
I suggest to change to simplify the macro usage.
Try to rewrite the macro in a way that allows to replace the line with _RESIZE_DIR(VERT, FWD)_